### DISCUSSION TITLE
How to populate form values dynamically?

### DISCUSSION THREAD

**Discussion by kostysh** [edmundhung/conform#367] [Q&A]
I need to fill all or specific form fields with values. These values cannot be passed to `defaultValue` (of `useForm` config) because these values will be loaded asynchronously while the user fills out the form.

**Comment by edmundhung** [comment]
Can you share more about your usecase? Something like an autocomplete with an async search?

**Reply by kostysh** [reply]
Hi, thank you for the response.
I have a form which includes a `dropdown` selector component with several options. When the user selects an option from the dropdown, I need to make a request to a remote API to fetch a specific value. Once the remote value is obtained, I must insert it into the `textarea` of the form, allowing the user to edit it as necessary.

**Reply by edmundhung** [reply]
You can make it a controlled input with the useState hook and update the state when needed.

**Reply by kostysh** [reply]
Yes, currently I using controlled input as a workaround. But this complicates the whole form handling. 

Why not have something like `setField('<name>', '<value>')` in the useForm hook?

Another use-case:
- Form already shown to the user
- There also exists a selector with options.
- When the user selects the option the form should be filled with the set of values related to the choice (that can be edited later).

What do you recommend? Using controlled inputs only and validating every input separately? I do not think so ((
